docs/modules/ROOT/pages/lua/guide/InterComputerCommunication.adoc

```diff
@@ -146,31 +146,31 @@ event.ignoreAll()
 event.listen(net)
 net:open(42)
 
+function foo(p1, p2, p3)
+ print(p1, p2, p3)
+end
+
 while true do
  local data = {event.pull()}
- e, s, sender, port, data = function(e,s,sender,port,...)
+ e, s, sender, port, data = (function(e,s,sender,port,...)
   return e, s, sender, port, {...}
- end(table.unpack(data))
+ end)(table.unpack(data))
  if e == "NetworkMessage" then
-  if d1 == "ping" then
+  if data[1] == "ping" then
    net:send(sender, port, "pong")
-  elseif d1 == "foo" then
+  elseif data[1] == "foo" then
    foo(table.unpack(data))
   end
  end
 end
-
-function foo(p1, p2, p3)
- print(p1, p2, p3)
-end
```

Comment:
Why the remove? I should visualize how the tuple result of table.unpack can be directly used as input for the function, this is not common Lua knowledge I wanted to show the people

Reply:
it isnt removed. 
Its moved to top cause the function part could never be reached as it is behind the while loop.
so you get an error "cant find function foo"

Reply:
afaik. Lua is executed line by line. So unless the function-line have been executed, it will not have been defined. This is more like php. I think this is a valid change.

Reply:
It depends :)
In an object oriented solution in Lua, PHP, Javascript aso. the whole object is parsed before execution. 
Means...  at the moment you create an instance of the object all methods inside the object are knowen before they are called from where ever, inside or outside the instance.
If you use non oo solution it is'nt. 
Both are possible in all of these Languages.